### PR TITLE
Allow skiping compression

### PR DESCRIPTION
**Pull request by GSKryachko**

### CODE REVIEW

File: internal/compression_structures_test.go [hunk 1]
```diff
@@ -155,3 +155,19 @@ func TestCompressAndEncryptError(t *testing.T) {
...
+		in := &testtools.BufCloser{Buffer: bytes.NewBufferString(tt.testString), Err: false}
+		compressed := internal.CompressAndEncrypt(in, nil, nil)
+
+		decompressed := &testtools.BufCloser{Buffer: &bytes.Buffer{}, Err: false}
+		_, err := decompressed.ReadFrom(compressed)
+		if err != nil {
+			t.Logf("%+v\n", err)
+		}
+
+		assert.Equalf(t, tt.testString, decompressed.String(), "compress: CascadeWriteCloser expected '%s' to be written but got '%s'", tt.testString, decompressed)
```
Comment (g0djan) [inline internal/compression_structures_test.go]: Too long line

File: internal/compression_structures_test.go [hunk 2]
```diff
@@ -155,3 +155,19 @@ func TestCompressAndEncryptError(t *testing.T) {
 		t.Errorf("compress: CompressingPipeWriter expected CompressAndEncryptError but got %v", re)
 	}
 }
+
+func TestCompressAndEncryptWithNoCompression(t *testing.T) {
+	for _, tt := range tests {
```
Comment (g0djan) [inline internal/compression_structures_test.go]: What means tt? Two twats?

File: internal/extract.go [hunk 3]
```diff
@@ -39,6 +39,13 @@ type UnsupportedFileTypeError struct {
 	error
 }
 
+type DecompressionError struct {
+	error
+}
+
+func newDecompressionError(err error) DecompressionError {
+	return DecompressionError{err}
+}
```
Comment (g0djan) [inline internal/extract.go]: Missed line. Maybe we should add formatter

File: internal/extract.go [hunk 4]
```diff
@@ -111,7 +117,11 @@ func decryptAndDecompressTar(writer io.Writer, readerMaker ReaderMaker, crypter
 			continue
 		}
 		err = decompressor.Decompress(writer, readCloser)
-		return errors.Wrapf(err, "DecryptAndDecompressTar: %v decompress failed. Is archive encrypted?", decompressor.FileExtension())
+		if err == nil {
+			return nil
+		}
+		decompressionError := newDecompressionError(err)
+		return errors.Wrapf(decompressionError, "DecryptAndDecompressTar: %v decompress failed. Is archive encrypted?", decompressor.FileExtension())
```
Comment (g0djan) [inline internal/extract.go]: Too long line

File: internal/compression_structures_test.go (hunk 2 above)
Comment (GSKryachko) [reply internal/compression_structures_test.go]: No idea - I've copypasted it from other test. I'll change it to something more understandable)

**Comment by GSKryachko** [comment]
I'll temporarily close the pr to not trigger CI when I do something in my branch.

**Pull request closed without merging**